x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/cache/common/CacheFileTests.java

```diff
@@ -455,13 +453,41 @@ public void testCacheFileCreatedAsSparseFile() throws Exception {
                 greaterThanOrEqualTo(cacheFile.getLength())
             );
 
-            if (Constants.LINUX) {
+            if (blockSize > 0L) {
                 final long nbBlocks = (cacheFile.getLength() + blockSize - 1) / blockSize; // ceil(cacheFile.getLength() / blockSize)
-                assertThat(
-                    "Cache file size mismatches (block size: " + blockSize + ", number of blocks: " + nbBlocks + ')',
-                    sizeOnDisk.getAsLong(),
-                    equalTo(nbBlocks * blockSize)
-                );
+                final long expectedSize = nbBlocks * blockSize;
+                if (blockSize == fourKb) {
+                    assertThat(
+                        "Cache file size mismatches (block size: "
+                            + blockSize
+                            + ", number of blocks: "
+                            + nbBlocks
+                            + ", file length: "
+                            + cacheFile.getLength()
+                            + ')',
+                        sizeOnDisk.getAsLong(),
+                        equalTo(expectedSize)
+                    );
+                } else {
+                    // block size other than usual default block size of 4KB indicates that a special filesystem may be at use; in this
+                    // case we verify that the allocated size for the file stays within some upper limit of an extra block of 4096L bytes
+                    assertThat(
+                        "Cache file size mismatches with non default block size (block size: "
+                            + blockSize
+                            + ", number of blocks: "
+                            + nbBlocks
+                            + ", file length: "
+                            + cacheFile.getLength()
+                            + ')',
+                        sizeOnDisk.getAsLong(),
+                        allOf(greaterThanOrEqualTo(expectedSize), lessThanOrEqualTo(expectedSize + fourKb))
+                    );
+                    assertThat(
+                        "Non default block size only used in test executed with encryption at rest",
+                        file.toAbsolutePath().toString(),
+                        startsWithIgnoringCase("/mnt/secret")
```

Comment:
This is not great but I can't figure a better way to check that encryption at rest is used.

See https://github.com/elastic/elasticsearch/blob/3f6fc8e592893deccb8bf231fbc01453149de116/.ci/jobs.t/elastic%2Belasticsearch%2Bperiodic%2Bear.yml#L28